### PR TITLE
[github] add label for filtering build errors as github issues

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Add new label in the automated github issues

## Why is it important?

Help with the filtering

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by mdelapenya — APPROVED** [review]

File: src/co/elastic/NotificationManager.groovy [hunk 1]
```diff
@@ -289,7 +289,7 @@ def createGitHubIssue(Map args = [:]) {
     arguments['archiveFile'] = false
     body = generateBuildReport(arguments)
   }
-  def labels = 'automation,ci-reported'
+  def labels = 'automation,ci-reported,ci-build'
```
Comment (mdelapenya) [inline src/co/elastic/NotificationManager.groovy:292]: What are the differences between `ci-build` and `ci-reported`? Could they be interchangeable?

**Review by cachedout — APPROVED** [review]

File: src/co/elastic/NotificationManager.groovy (hunk 1 above)
Comment (v1v) [reply src/co/elastic/NotificationManager.groovy:292]: `ci-reported` is the generic one we have used for the automation regardless of the test-flakiness or build issues. While `ci-build` is the one to filter build related failures (regardless of what the build failure is about)

The idea is to provide some unique github label so people can use queries to filter the kind of the GitHub issues. At the moment is all together

Comment (cachedout) [reply src/co/elastic/NotificationManager.groovy:292]: I see. What if we went the other direction and specifically made a label for flakiness and then a second label for build failures?

Comment (v1v) [reply src/co/elastic/NotificationManager.groovy:292]:
```suggestion
  def labels = 'automation,ci-reported,build-failures'
```

File: src/test/groovy/NotificationManagerStepTests.groovy [hunk 2]
```diff
@@ -1063,7 +1063,7 @@ class NotificationManagerStepTests extends ApmBasePipelineTest {
     script.createGitHubIssue(comment: 'my build report')
     printCallStack()
     assertFalse(assertMethodCallContainsPattern('githubCreateIssue', 'assignee'))
-    assertTrue(assertMethodCallContainsPattern('githubCreateIssue', 'labels=automation,ci-reported}'))
+    assertTrue(assertMethodCallContainsPattern('githubCreateIssue', 'labels=automation,ci-reported,ci-build}'))
```
Comment (v1v) [inline src/test/groovy/NotificationManagerStepTests.groovy:1066]:
```suggestion
    assertTrue(assertMethodCallContainsPattern('githubCreateIssue', 'labels=automation,ci-reported,build-failures}'))
```

File: src/test/groovy/NotificationManagerStepTests.groovy [hunk 3]
```diff
@@ -1084,7 +1084,7 @@ class NotificationManagerStepTests extends ApmBasePipelineTest {
   void test_createGitHubIssue_with_labels() throws Exception {
     script.createGitHubIssue(githubLabels: 'foo', comment: 'my build report')
     printCallStack()
-    assertTrue(assertMethodCallContainsPattern('githubCreateIssue', 'labels=automation,ci-reported,foo}'))
+    assertTrue(assertMethodCallContainsPattern('githubCreateIssue', 'labels=automation,ci-reported,ci-build,foo}'))
```
Comment (v1v) [inline src/test/groovy/NotificationManagerStepTests.groovy:1087]:
```suggestion
    assertTrue(assertMethodCallContainsPattern('githubCreateIssue', 'labels=automation,ci-reported,build-failures,foo}'))
```

**Pull request merged**